### PR TITLE
remote-files: expand and improve remote Ignition file section

### LINKED ISSUE #430: Improve remote-ign docs
We should discourage plain HTTP.  Let's switch to HTTPS and give examples of `gs` and `s3` URLs.

### PR DESCRIPTION
**Pull request by Okeanos**
Convert basic examples to more secure HTTPS and separately explain plain HTTP with an added warning.

Add hint about other supported protocols and an HTTPS proxy example to expand existing documentation with other supported (common) use cases.

Resolves #430

### CODE REVIEW

File: modules/ROOT/pages/remote-ign.adoc [hunk 1]
```diff
@@ -2,17 +2,19 @@
...
 ----
 variant: fcos
 version: 1.4.0
 ignition:
   config:
     replace:
-      source: http://example.com/sample.ign
+      source: https://example.com/sample.ign
       verification:
         hash: sha512-e2bb19fdbc3604f511b13d66f4c675f011a63dd967b97e2fe4f5d50bf6cb224e902182221ba0f9dd87c0bb4abcbd2ab428eb7965aa7f177eb5630e7a1793e2e6
```
Comment (bgilbert) [inline modules/ROOT/pages/remote-ign.adoc:19]: In practice, `verification` isn't used much with HTTPS.  I'd be inclined to use it only in the HTTP example.

File: modules/ROOT/pages/remote-ign.adoc [hunk 2]
```diff
@@ -55,3 +57,42 @@ passwd:
       ssh_authorized_keys:
         - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDHn2eh...
 ----
+
+Retrieving remote Ignition files via plain HTTP is also possible as shown below.
+
+WARNING: Retrieving a remote Ignition file via HTTP is discouraged for obvious reasons. When using HTTP, it is advisable to use the verification option to ensure the contents haven't been tampered with.
```
Comment (bgilbert) [inline modules/ROOT/pages/remote-ign.adoc:63]: Obvious or not, let's spell it out:

```suggestion
WARNING: Retrieving a remote Ignition config via HTTP exposes the contents of the config to anyone monitoring network traffic. When using HTTP, it is advisable to use the verification option to ensure the contents haven't been tampered with.
```

File: modules/ROOT/pages/remote-ign.adoc [hunk 3]
```diff
@@ -55,3 +57,42 @@ passwd:
...
+version: 1.4.0
+ignition:
+  config:
+    replace:
+      source: http://example.com/sample.ign
+      verification:
+        hash: sha512-e2bb19fdbc3604f511b13d66f4c675f011a63dd967b97e2fe4f5d50bf6cb224e902182221ba0f9dd87c0bb4abcbd2ab428eb7965aa7f177eb5630e7a1793e2e6
+----
+
+In case you need to retrieve a remote Ignition file but have no direct access to the remote host you can specify a proxy for both plain HTTP and HTTPS. Hosts that should be excluded from proxying can also be specified.
```
Comment (bgilbert) [inline modules/ROOT/pages/remote-ign.adoc:78]:
```suggestion
If you need to retrieve a remote Ignition file but have no direct access to the remote host, you can specify a proxy for plain HTTP and/or HTTPS. You can also specify hosts that should be excluded from proxying.
```

**Review by bgilbert — COMMENTED** [review]
Looks good generally.  Thanks!

File: modules/ROOT/pages/remote-ign.adoc (hunk 2 above)
Comment (Okeanos) [reply modules/ROOT/pages/remote-ign.adoc]: Applied your wording fixes.

File: modules/ROOT/pages/remote-ign.adoc (hunk 1 above)
Comment (Okeanos) [reply modules/ROOT/pages/remote-ign.adoc]: Removed `verification` from all but the plain HTTP example.

File: modules/ROOT/pages/remote-ign.adoc (hunk 3 above)
Comment (Okeanos) [reply modules/ROOT/pages/remote-ign.adoc]: Applied your wording fixes.

**Review by bgilbert — APPROVED** [review]
Thanks!

**Pull request merged**